### PR TITLE
Squash migrations / fix migrations for nautobot 2.0

### PR DESCRIPTION
**Pull request by gsnider2195**
### NOTE: This was the result of trying almost every method to get these migrations squashed / working with 2.0 and this was the only option that worked.

We did not squash migrations `0002` or `0003` because they contained data migrations that still need to run in 2.0.

We also had to make some updates to the existing migrations to fix some circular dependency issues that showed up after the squash. Part of this was adding migration `0008` which should be a no-op for any installs that have already upgraded to 2.0.

The `run_before` for the initial migration `0001` is no longer necessary because if that migration hasn't run yet, it will be skipped in favor of the squashed migration `0101`. So we removed it.

We also updated the `run_before` logic for the `0004`/`0005` migrations to fix circular dependency issues when squashing migrations that contained `run_before`

### CODE REVIEW

File: nautobot_device_onboarding/migrations/0101_squashed.py
Comment (glennmatthews) [inline nautobot_device_onboarding/migrations/0101_squashed.py:1]: Is numbering this one as 0101 confusing? Can we number it as something like 0001_0004_0005_0006 instead and would that be clearer?

File: nautobot_device_onboarding/migrations/0005_migrate_site_to_location_part_3.py [hunk 1]
```diff
@@ -16,3 +16,13 @@ class Migration(migrations.Migration):
             name="site",
         )
     ]
+
+    def __init__(self, name, app_label):
```
Comment (glennmatthews) [inline nautobot_device_onboarding/migrations/0005_migrate_site_to_location_part_3.py:20]: Mind adding a docstring or comment explaining this and the similar logic in the previous file for future reference?

File: nautobot_device_onboarding/migrations/0101_squashed.py
Comment (jathanism) [reply nautobot_device_onboarding/migrations/0101_squashed.py:1]: I'd be concerned about having another `0001_` prefixed file that would be adjacent to `0001_initial`, but I agree with the sentiment of naming the migration more verbosely.

**Comment by scetron** [comment]
I'd also take some suggestions on what we might need to do for the existing release? Should that be yanked or anything?

**Comment by glennmatthews** [comment]
Gary can confirm but I don't think the existing release needs to be yanked. We'll just want a new patch release cut ASAP after this fix gets merged.

**Comment by gsnider2195** [comment]
This patch will be compatible with the existing release. If someone has tried to upgrade and it failed migration, they should be able to upgrade to the release with these migrations and run migrations successfully.

**Review by jathanism — APPROVED** [review]
🚢 🇮🇹

**Review by scetron — APPROVED** [review]
Looks good to me! Thanks very much for this.

**Pull request merged**